docs/plugins/ingest.asciidoc

```diff
@@ -403,57 +403,110 @@ An example that adds the parsed date to the `timestamp` field based on the `init
+    "field": "my_field"
+    "value": 582.1
+  }
 }
 --------------------------------------------------
 
-The following metadata attributes can be modified in this processor: `_index`, `_type`, `_id`, `_routing`, `_parent`,
-`_timestamp` and `_ttl`. All these metadata attributes can be specified in the body of the `meta` processor.
+On top of this fields from the source are always accessible via the `_source` prefix:
+
+[source,js]
+--------------------------------------------------
+{
+  "set": {
+    "field": "_source.my_field"
+    "value": 582.1
+  }
+}
+--------------------------------------------------
 
-Also the metadata settings in this processor are templatable which allows metadata field values to be replaced with
-field values in the source of the document being indexed. The mustache template language is used and anything between
-`{{` and `}}` can contain a template and point to any field in the source of the document.
+Metadata fields can also be accessed in the same way as fields from the source. This
+is possible because Elasticsearch doesn't allow fields in the source that have the
+same name as metadata fields.
 
-The following example documents being processed end up being indexed into an index based on the resolved city name by
-the `geoip` processor. (for example `city-amsterdam`)
+The following example sets the id of a document to `1`:
 
 [source,js]
 --------------------------------------------------
 {
-  "description" : "...",
-  "processors" : [
-    {
-      "geoip" : {
-        "source" : "ip"
-      }
-    },
-    {
-      "meta" : {
-        "_index" : "city-{{geoip.city_name}}"
-      }
-    }
-  ]
+  "set": {
+    "field": "_id"
+    "value": "1"
+  }
+}
+--------------------------------------------------
+
+The following metadata fields are accessible by a processor: `_index`, `_type`, `_id`, `_routing`, `_parent`,
+`_timestamp` and `_ttl`.
+
+Beyond metadata fields and source fields, the ingest plugin also adds ingest metadata to documents being processed.
+These metadata properties are accessible under the `_ingest` key. Currently the ingest plugin adds the ingest timestamp
+under `_ingest.timestamp` key to the ingest metadata, which is the time the ingest plugin received the index or bulk
+request to pre-process. But any processor is free to add more ingest related metadata to it.
+
+The following example adds a field with the name `received` and the value is the ingest timestamp:
+
+[source,js]
+--------------------------------------------------
+{
+  "set": {
+    "field": "received"
+    "value": "{{_ingest.timestamp}}"
+  }
+}
+--------------------------------------------------
+
+As opposed to Elasticsearch metadata fields, the ingest metadata field name `_ingest` isn't restricted by Elasticsearch
+and is a valid field name in the source of a document, so there is a small mismatch.
+In the ingest plugin when is referred to `_ingest`  in a field setting or template then ingest metadata is accessed.
```

Comment:
Looking to simplify this block: As opposed to Elasticsearch metadata fields, the ingest metadata field name `_ingest` can be used as a valid field name in the source of a document. Use `_source._ingest` to refer to it, otherwise `_ingest` will be interpreted as ingest metadata fields by the ingest plugin.